### PR TITLE
Feature vision improv - Tuning and Image Saving

### PR DESCRIPTION
**Pull request by gilben1**
This pull request is a bit of a back-fill since multiple cards have been completed under this request.

Cards this pull request addresses: 
Vision -- Tune and save data for given red ball
Vision -- Figure out how to take discrete pictures from stream

To test:

    Pull feature_vision_improv
    Checkout master
    Merge feature_vision_improv in
    Catkin make
    Test with robot

If you review this PR, please add your name and email below.

Reviewers:
Reviewed-by: Alice Michael <malice@pdx.edu>
Reviewed-by: Mel Flygare <mflygare@pdx.edu>

### CODE REVIEW

**Comment by Sappytomb796** [comment]
This looks good to me in terms of the code and consider this a +1.  
I just want to make sure I understand in terms of our project.  Once this is merged and uploaded to the robot, can we demo that it tracks the ball now? aka is that the expected outcome or is there still work to make sure that it's all integrated into the soccer mode that I didn't realize?

**Comment by gilben1** [comment]
> Once this is merged and uploaded to the robot, can we demo that it tracks the ball now? aka is that the expected outcome or is there still work to make sure that it's all integrated into the soccer mode that I didn't realize?

That first commit changes those parameter values that are currently utilized by the existing ball detection. It should detect in the existing soccer mode, yes, and was tested to ensure it did so.

File: op3_ball_detector/src/ball_detector.cpp [hunk 1]
```diff
@@ -342,6 +343,23 @@ bool BallDetector:: getParamCallback(op3_ball_detector::GetParameters::Request &
...
 
+// Callback for image saving
+bool BallDetector::saveImageCallback(op3_ball_detector::SaveImage::Request &req, op3_ball_detector::SaveImage::Response &res)
+{
+  std::string filename = "/home/robotis/" + req.params.name + ".png";
+  if (cv_img_ptr_sub_ != NULL) {
+    if (cv::imwrite(filename, cv_img_ptr_sub_->image) == false) {
+      res.returns.name = "Failed for some reason";
+    }
+    res.returns.name = "Saved";
```
Comment (mel-flygare96) [inline op3_ball_detector/src/ball_detector.cpp]: Wouldn't this overwrite line 352 or am I misunderstanding something?

Comment (gilben1) [reply op3_ball_detector/src/ball_detector.cpp]: 9626c6d fixes the omitted else statement.

**Comment by BradyT98** [comment]
I don't see anything in here that looks terribly wrong. I think it is good to merge.

**Review by mel-flygare96 — APPROVED** [review]
Looks good.

**Pull request merged**